### PR TITLE
Meta performance

### PR DESCRIPTION
**Pull request by oliver-s-lee**
Adds support for number of CPUs and memory used in the calculation in two new metadata keys:
 - `metadata['num_cpus']` - Number of parallel CPUs, integer
 - `metadata['memory']` - Max allowed memory usage in bytes (total, across all CPUs)

WIP for now because we're missing tests

### CODE REVIEW

**Comment by shivupa** [comment]
An acceptable test could just be making sure that the attributes are present for say `dvb_sp` tests. I think this is a nice bite size PR and we could have further improvements be other PRs? or did you want this for other parsers as well?

**Comment by oliver-s-lee** [comment]
No no I was thinking of leaving it just for these 3 for now, they are my favourites  anyway :)

I think it shouldn't be too hard to check for actual numbers of CPUs/memory, I'll have a look later this week

**Comment by oliver-s-lee** [comment]
Split memory into `metadata['memory_available']`, which corresponds to the amount of memory requested in the input file, and `metadata['memory_used']` which is the max amount of memory used, as reported by the program (if available).

File: data/Gaussian/basicGaussian16/dvb_gopt.cjson
Comment (shivupa) [inline data/Gaussian/basicGaussian16/dvb_gopt.cjson:1]: this file doesn't need to be in this right?

**Review by shivupa — CHANGES_REQUESTED** [review]
1 change then Good to merge

File: data/Gaussian/basicGaussian16/dvb_gopt.cjson
Comment (oliver-s-lee) [reply data/Gaussian/basicGaussian16/dvb_gopt.cjson:1]: oops, that's what you get for bulk adding folders, deleted

**Review by shivupa — APPROVED** [review]

**Pull request merged**